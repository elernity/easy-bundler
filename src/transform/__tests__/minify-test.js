'use strict';
require('babel-register')();
const minify = require('../minify');
const minifierConfig = require('../../config/default').systemConfig.minifierConfig;

describe('代码混淆测试', () => {
    it('应混淆代码', () => {
        const minifiedCode = minify('for(let index = 0; index < 1000; index++) {console.log(index);}', minifierConfig);
        expect(minifiedCode).toMatchSnapshot();
    });

    it('应抛错，代码无法解析', () => {
        expect(() => minify('do(', minifierConfig)).toThrow();
    });
});
