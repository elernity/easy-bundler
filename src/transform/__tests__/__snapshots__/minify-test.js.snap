// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`代码混淆测试 应混淆代码 1`] = `"for(let o=0;o<1e3;o++)console.log(o);"`;
